### DISCUSSION TITLE
Reduce unused JavaScript

### DISCUSSION THREAD

**Discussion by theguriev** [nuxt/nuxt#19982] [Questions] [answered]
Can someone tell me how to figure out what exactly needs to be thrown out of the entry.js? Judging by pagespeed I have more than half of the file not used. 

<img width="958" alt="image" src="https://user-images.githubusercontent.com/8502021/228541296-c452b7bb-765c-438a-ae55-ed620422e026.png">

Perhaps there is already an article somewhere on how to optimize Nuxt 3 to achieve this. Treeshaking doesn't really seem to work for me.

**Comment by TheAlexLichter** [comment, marked as answer]
Just because you don't use some of the JS, it still can be useful for e.g. error handling, parts of Vue or the router that you don't use or similar.

What you can always do is running `nuxi analyze` to check your bundle and see what dependencies are too big.

**Reply by theguriev** [reply]
Thank you. I totally forgot about `nuxi analyze`. It helped. At least now I know what might be refactored.
